File: _DSAPractice/0053_MaximumSubarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        mx = max(nums)
        sm = 0
        for i in range(len(nums)):
            sm+=nums[i]
            mx= max(mx,sm)
            sm=max(sm,0)
        return mx
            
        

File: _DSAPractice/0053_MaximumSubarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int n=nums.length,max=nums[0],prev=nums[0];
        for(int i=1;i<n;i++){
            prev= nums[i]+Math.max(prev,0);
            max=Math.max(max,prev);
        }
        return max;
    }
}
